runtime/compiler/rpc/StreamTypes.hpp

```diff
@@ -48,7 +52,16 @@ class StreamFailure: public virtual std::exception
 class StreamCancel: public virtual std::exception
    {
 public:
-   virtual const char* what() const throw() { return "compilation canceled by client"; }
+   StreamCancel(J9ServerMessageType type, uint64_t clientId=0) : _type(type), _clientId(clientId) { }
```

Comment:
`_clientID` is used for `terminateClient()` in a catch block.

Reply:
Type is to indicate that there are two types of `StreamCancel`:
1. client aborts compilation
2. client is exiting
We handle these two types differently, not only used to decide what message to print.
I put them both under `StreamCancel` because they are both under the category where client cancels the compilation.